ocs_ci/ocs/cluster_load.py

```diff
@@ -193,89 +192,114 @@ def reach_cluster_load_percentage(self):
-                    f"\nthe given {time_to_wait} seconds timeout. Breaking"
+                    wrap_msg(
+                        "Could not determine the cluster IOPS limit within"
+                        f"the given {time_to_wait} seconds timeout. Breaking"
+                    )
                 )
-                limit_reached = True
+                break
 
-            cluster_used_space = get_percent_used_capacity()
-            if cluster_used_space > 60:
+            elif cluster_used_space > 60:
                 logger.warning(
-                    f"Cluster used space is {cluster_used_space}%. Could "
-                    "not reach the cluster IOPS limit before the "
-                    "used spaced reached 60%. Breaking"
+                    wrap_msg(
+                        f"Cluster used space is {cluster_used_space}%. Could "
+                        "not reach the cluster IOPS limit before the "
+                        "used spaced reached 60%. Breaking"
+                    )
                 )
-                limit_reached = True
+                break
 
         self.cluster_limit = cluster_limit
         logger.info(wrap_msg(f"The cluster IOPS limit is {self.cluster_limit:.2f}"))
-        logger.info(
-            f"Deleting all DC FIO pods that have FIO rate parameter of {rate}"
-        )
+        logger.info(f"Deleting all DC FIO pods that have large FIO rate")
         while self.dc_objs:
             self.decrease_load(wait=False)
 
+        target_iops = self.cluster_limit * self.target_percentage
         # Creating the first pod of small FIO 'rate' param, to speed up the process.
         # In the meantime, the load will drop, following the deletion of the
         # FIO pods with large FIO 'rate' param
-        rate = '15M'
         logger.info(
-            f"Creating FIO pods with a rate parameter of {rate}, one by "
-            "one, until the target percentage is reached"
+            f"Creating FIO pods, one by one, until the target percentage is reached"
         )
-        self.increase_load(rate=rate)
-        target_iops = self.cluster_limit * self.target_percentage
-        self.current_iops = self.get_query(query=constants.IOPS_QUERY)
+        self.increase_load_and_print_data()
         msg = (
             f"The target load, in IOPS, is: {target_iops}, which is "
             f"{self.target_percentage*100}% of the {self.cluster_limit} cluster limit"
         )
         logger.info(wrap_msg(msg))
 
-        while self.current_iops < target_iops * 0.95:
-            wait = False if self.current_iops < target_iops / 2 else True
-            self.increase_load_and_print_data(rate=rate, wait=wait)
+        if target_iops < 400:
+            allowed_diff = 0.82
+            wait_factor = 0.4
+        elif 400 <= target_iops <= 800:
+            allowed_diff = 0.85
+            wait_factor = 0.5
+        elif 800 <= target_iops <= 1200:
+            allowed_diff = 0.88
+            wait_factor = 0.6
```

Comment:
You can use RangeKeyDict:
require: "pip install range-key-dict"
Example:
range_key_dict = RangeKeyDict({
(400, 800):(0.85,0.5),
(800, 1200):(0.88,0.6)
})

Reply:
I think it would be better to stick to the if conditions because I don't think this worths adding one more module to the setup.py requirements and also, I couldn't find a way to specify a range without an upper limit, that I need in https://github.com/red-hat-storage/ocs-ci/pull/2612/files#diff-198938419d5d1b9b803f1d40b020c0abR296

Thanks for the suggestion, it looks very neat!

Reply:
discussed offline. Will use `math.inf` for the upper limit condition